commands/http/handler.go

```diff
@@ -232,7 +234,9 @@ func sendResponse(w http.ResponseWriter, r *http.Request, res cmds.Response, req
 	if _, ok := res.Output().(io.Reader); ok {
 		// set streams output type to text to avoid issues with browsers rendering
 		// html pages on priveleged api ports
-		mime = "text/plain"
+		if mime != mimeTypes[cmds.Tar] && mime != mimeTypes[cmds.Gzip] {
```

Comment:
maybe can make it something like:

```go
if !xssSafeMimeType(mime) {
  mime = mimeTypes[cmds.Text]
}
```

and elsewhere something like:

```go
var xssSafeMimeTypes = []string{
  mimeTypes[cmds.Tar],
  mimeTypes[cmds.GZip],
  mimeTypes[cmds.Text],
  mimeTypes[cmds.JSON],
}

func xssSafeMimeType(mime string) {
  for _, t := range xssSafeMimeTypes {
    if t == mime {
      return true
    }
  }
  return false
}
```
to make it clear _why_ this is happening, and easier to broaden the safety boundary, in a single place.

this probably could be a separate package, and may already exist somewhere.